skimage/measure/fit.py

```diff
@@ -730,14 +936,27 @@ def ransac(
         ``is_model_valid(model, *random_data)`` and
         ``is_data_valid(*random_data)`` must all take each data array as
         separate arguments.
-    model_class : object
-        Object with the following object methods:
+    model_class : type
+        Class with the following methods:
 
-         * ``success = estimate(*data)``
-         * ``residuals(*data)``
+        * Either:
+
+          * ``from_estimate`` class method returning transform instance, as in
+            ``tform = model_class.from_estimate(*data)``; the resulting
+            ``tform`` should be truthy (``bool(tform) == True``) where
+            estimation succeeded, or falsey (``bool(tform) == False``) where it
+            failed;  OR
+          * (deprecated) ``estimate`` instance method, returning flag to
+            indicate successful estimation, as in ``tform = model_class();
+            success = tform.estimate(*data)``. ``success == True`` when
+            estimation succeeded, ``success == False`` when it failed.
+
+        * ``residuals(*data)``
+
+        See :class:`RansacModelProtocol` for the protocol definition.  In
+        general, ensure that `model_class` is an instance of
+        ``RansacModelProtocol`` (this checks it conforms to the protocol).
```

Comment:
Users who are not familiar with Protocols (and many aren't) may be confused by this.
Perhaps we can give them clearer directions: "Your model should conform to (implement all of the methods / attributes specified by) RansacModelProtocol. An easy check to see whether that is the case is `isinstance(MyModel, RansacModelProtocol)`."

I'm not sure if the above is the correct way to check; is it?

Reply:
It is.  I've expanded the text a bit.